### DISCUSSION TITLE
Broken pipe error when rendering quarto doc in GitHub actions

### DISCUSSION THREAD

**Discussion by boshek** [quarto-dev/quarto-cli#7219] [Q&A] [answered]
### Description

I am getting this funny error with a self-hosted runner in GitHub actions. I am trying to create a public example of this but I just wanted to post this here in case anyone has any thoughts. This happens when I try to run `quarto render` in GHA. I think the R error is a red herring and somehow the connection is being broken resulting in the broken pipe error. This _easily_ could be a GHA issue and not a quarto issue but I was just wondering if there were any quarto specific troubleshooting steps anyone could suggest. Will work on seeing if I can replicate in a shareable space. 

```
ERROR: Error executing '/usr/bin/Rscript': Broken pipe (os error 32)
Error in if (n == 1) word else paste(word, "s", sep = "") : 
  the condition has length > 1
Calls: source ... sprintf -> <Anonymous> -> nplural -> paste -> plural
Execution halted

Unable to locate an installed version of R.
Install R from https://cloud.r-project.org/

ERROR: Error
    at renderFiles (file:///opt/quarto/bin/quarto.js:86658:30)
    at async renderProject (file:///opt/quarto/bin/quarto.js:86800
    at async Command.fn (file:///opt/quarto/bin/quarto.js:90856:32)
    at async Command.execute (file:///opt/quarto/bin/quarto.js:8437:13)
    at async quarto (file:///opt/quarto/bin/quarto.js:127545:5)
    at async file:///opt/quarto/bin/quarto.js:127563:9
Error: Process completed with exit code 1.
```

**Comment by mcanouil** [comment]
Could you share a small self-contained "working" (reproducible) example to work with, _i.e._, a complete Quarto document or a Git repository? Thanks.

Also what is the version or Quarto you are using?
What's the setup of your runner?

Troubleshooting guide: <https://quarto.org/docs/troubleshooting/index.html>

Note that the error is telling you that "n" is not of length one which is an error in latest R version.
`if (n == 1) word else paste(word, "s", sep = "")` has nothing to do with Quarto internal R code, so likely your code is faulty (or maybe an R package).

You can share a Quarto document using the following syntax, _i.e._, using more backticks than you have in your document (usually four ` ```` `).

`````md
````qmd
---
title: "Reproducible Quarto Document"
format: html
---

This is a reproducible Quarto document using `format: html`.
It is written in Markdown and contains embedded R code.
When you run the code, it will produce a plot.

```{r}
plot(cars)
```

The end.
````
`````

**Comment by cderv** [comment]
> ```
> Error in if (n == 1) word else paste(word, "s", sep = "") : 
>  the condition has length > 1
> Calls: source ... sprintf -> <Anonymous> -> nplural -> paste -> plural
> ```

This seems to be a real issue in rendering your project. 

I believe this comes from **renv** : See their source code
https://github.com/rstudio/renv/blob/36e1b5192fbc9f380dd4fffc6ee837856d5d3e75/R/utils.R#L243-L249

So somehow there is a problem. You should check your **renv** version, or maybe try without **renv** in your project to see how it behaves.

**Reply by cderv** [reply, marked as answer]
Also it seems you are not the only one with this 
* https://github.com/rstudio/renv/issues/1655

It seems it was a CI configuration there

**Reply by boshek** [reply]
That @cderv for this helpful answer. I have not seen that renv issue. Surely the issue. Unfortunately on my side this happens intermittently. I suspect it is related to resource allocation at the same when renv is looking for packages. If resources happen to be available, everything is fine but this bonks with this error when it is not. I've tried ignore the ci file as suggested in that issue. So far so good!

**Comment by robpetrosino** [comment]
Hello, I am also encountering the same issue for some reason -- is there any solution?

**Reply by cderv** [reply]
Hi, 

If this is the exact same issue, did you look at solution selected at https://github.com/quarto-dev/quarto-cli/discussions/7219#discussioncomment-7267388

This issue was related to **renv**. If this is different for you, please do open a new thread with information about your context to discuss. 

Thank you !
